### PR TITLE
Drydeposition 

### PR DESCRIPTION
**Pull request by rosiealice**
### Description of changes
Does not update the science parts of the season_index calculation, yet.. 
To pair with forthcoming FATES-side changes. 

### Specific notes

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):

Are answers expected to change (and if so in what way)?

Any User Interface Changes (namelist or namelist defaults changes)?

Does this create a need to change or add documentation? Did you do so?

Testing performed, if any:
(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on derecho for intel/gnu and izumi for intel/gnu/nag/nvhpc is the standard for tags on master)

**NOTE: Be sure to check your coding style against the standard
(https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines) and review
the list of common problems to watch out for
(https://github.com/ESCOMP/CTSM/wiki/List-of-common-problems).**

### CODE REVIEW

**Review by mvertens — APPROVED** [review]

**Comment by mvertens** [comment]
@rosiealice - thanks for this. Can you summarize what test(s) you ran with this - and what code base? That would be super helpful. Actually - looking at the PR changes - I'm not sure this is needed for this PR - but its good practice to include this.
@mvdebolskiy  - I have reviewed these changes and approved. Do you want to want to review as well - or should I just merge?

File: src/biogeochem/DryDepVelocity.F90 [hunk 1]
```diff
@@ -376,12 +381,12 @@ subroutine depvel_compute( bounds, &
             ! 4 - Winter, snow on ground and subfreezing
             ! 5 - Transitional spring with partially green short annuals
 
-
-            !mlaidiff=jan-feb
-            minlai=minval(annlai(:,pi))
-            maxlai=maxval(annlai(:,pi))
-
-            index_season = -1
+            if(.not. use_fates)then
```
Comment (mvdebolskiy) [inline src/biogeochem/DryDepVelocity.F90]: Why this conditions is removed?

File: src/biogeochem/DryDepVelocity.F90 [hunk 2]
```diff
@@ -399,10 +404,14 @@ subroutine depvel_compute( bounds, &
...
-            else if(elai(pi) > 0.5_r8*maxlai) then
+            else if(.not.use_fates .and. elai(pi) > 0.5_r8*maxlai) then
                index_season = 1
             endif
 
+            if(use_fates)then
+              index_season = canopystate_inst%drydep_season_patch(pi)              
+            endif
+
             if (index_season<0) then
```
Comment (mvdebolskiy) [inline src/biogeochem/DryDepVelocity.F90]: There is .not. use_fates above, however, it is not here.
This can possibly overwrite season index.

File: src/biogeophys/CanopyStateType.F90 [hunk 3]
```diff
@@ -52,6 +52,9 @@ module CanopyStateType
 
      real(r8) , pointer :: dleaf_patch              (:)   ! patch characteristic leaf width (diameter) [m]
                                                           ! for non-ED/FATES this is the same as pftcon%dleaf()
+     integer ,  pointer :: wesley_pft_index_patch   (:)   ! Wesley PFT index for FATES drydep calculations
+     integer ,  pointer :: drydep_season_patch      (:)   ! Season for dry deposition calculations
```
Comment (mvdebolskiy) [inline src/biogeophys/CanopyStateType.F90]: Can you name it `wesley_veg_index_patch` and `wesley_season_index_patch` for consistency? And also in comments have `dry deposition` spelled out everywhere?

File: src/utils/clmfates_interfaceMod.F90 [hunk 4]
```diff
@@ -1608,7 +1610,8 @@ subroutine wrap_update_hlmfates_dyn(this, nc, bounds_clump,      &
           z0m(col%patchi(c)+1:col%patchf(c)) = 0.0_r8
           displa(col%patchi(c)+1:col%patchf(c)) = 0.0_r8
           dleaf_patch(col%patchi(c)+1:col%patchf(c)) = 0.0_r8
-
+          wesley_pft_index_patch(col%patchi(c)+1:col%patchf(c)) = 0
```
Comment (mvdebolskiy) [inline src/utils/clmfates_interfaceMod.F90]: Can you do `0.0_r8` here, like it is done everywhere?

**Review by mvdebolskiy — CHANGES_REQUESTED** [review]

**Comment by mvdebolskiy** [comment]
One of my issues is that with `use_fates` conditions, which includes`use_fates_sp`, there will be different behaviour for SP mode than it was before.

File: src/utils/clmfates_interfaceMod.F90 (hunk 4 above)
Comment (mvdebolskiy) [reply src/utils/clmfates_interfaceMod.F90]: In DryDepVelocity the initital valaue is `-1` and there are checks `<0`.

**Comment by mvdebolskiy** [comment]
Also, Can you add checks for valid index values either in DryDepVelocity or in clmfates_interfaceMod where values from fates are passed? There should be an endrun with `Wesley ... index out of bounds: <values> <valid_range>`.

**Comment by rosiealice** [comment]
Hi @mvertens. Maybe hold off on merging it yet. I am still testing the FATES side changes, so this is not yet finished...

**Comment by mvertens** [comment]
@rosiealice - would be good to have this in draft form if you are still testing. And I've been talking to @mvdebolskiy and we feel that we should have a pared down test suite for betzy on our end that is always run as part of PRs. Let's talk more at tomorrow's meeting.

**Comment by rosiealice** [comment]
Sure. I pressed the 'convert to draft' button. Was that right? :)

**Comment by mvertens** [comment]
Perfect!

File: src/biogeophys/CanopyStateType.F90 (hunk 3 above)
Comment (rosiealice) [reply src/biogeophys/CanopyStateType.F90]: Sure. Agree that would add better name spacing symmetry...

File: src/biogeochem/DryDepVelocity.F90 (hunk 1 above)
Comment (rosiealice) [reply src/biogeochem/DryDepVelocity.F90]: Do you mean minlai and maxlai? (they are now below in the if statement to make the logic clearer)...

File: src/utils/clmfates_interfaceMod.F90 (hunk 4 above)
Comment (rosiealice) [reply src/utils/clmfates_interfaceMod.F90]: These are defined at integers in Canopystatemod so I think they should just be '0'? 

Would it be good form to start this as the unset values of -1? Or just stick to 'zaeroing'?

**Comment by rosiealice** [comment]
@mvdebolskiy I changed the logic here
https://github.com/rosiealice/ctsm/blob/2e4293d1281dbac04684488e75b4c188c8387df0/src/main/clm_initializeMod.F90#L707
so that the .not.use_fates statement is a different if statement to the use_fates_sp. Does that fix the concern you voiced earlier? 

if ( n_drydep > 0 .and. .not. use_fates ) then
       call readAnnualVegetation(bounds_proc, canopystate_inst)
       ! Call interpMonthlyVeg for dry-deposition so that mlaidiff will be calculated
       ! This needs to be done even if FATES, CN or CNDV is on!
       call interpMonthlyVeg(bounds_proc, canopystate_inst)
    endif 
    if ( use_fates_sp ) then
       ! If fates has satellite phenology enabled, get the monthly veg values
       ! prior to the first call to SatellitePhenology()
       call interpMonthlyVeg(bounds_proc, canopystate_inst)
    end if

**Comment by rosiealice** [comment]
I made paired CLM-SP and FATES-SP simulations to compare the dry deposition outputs. 

The CLM-SP simulation is here: 
/cluster/work/users/rosief/git/NorESM_alpha07test/cime/scripts/CLM6-SP.n1850.ne30_tn14.drydep

and the FATES-SP case is here:
(I have notes on how I created these, jut need to think of a place to put them) 
/cluster/work/users/rosief/git/NorESM_drydep_test/cime/scripts/FATES-SP.n1850.ne30_tn14.drydep

I ran these for three years each with -drydep specified by
./xmlchange CLM_BLDNML_OPTS="-drydep" --append

 This is a random month of output run through my crude plotting script (working on getting these into a diagnostics package is on our to do list). 

The differences are very small (suspiciously so??!) but nonetheless, here they are. 

As for the MEGAN PR, I think this code is ready for running through tests. @mvdebolskiy @mvertens 

### Plotted by gridcell
drydep_DSTDEP
drydep_BCDEP
drydep_OCDEP



### Plotted by latitude
drydep_lat_DSTDEP
drydep_lat_BCDEP
drydep_lat_OCDEP

### Only the differences plotted (to check they are not all zeros) 
drydep_diffs_OCDEP
drydep_diffs_DSTDEP
drydep_diffs_BCDEP

**Comment by rosiealice** [comment]
Closed as replaced by #106
